scalyr_agent/tests/test_util_test.py

```diff
@@ -0,0 +1,59 @@
+# Copyright 2014 Scalyr Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------------------
+#
+#
+
+from __future__ import absolute_import
+
+import os
+
+from scalyr_agent.test_base import BaseScalyrLogCaptureTestCase
+
+__all__ = [
+    'LogCaptureClassTestCase'
+]
+
+
+class LogCaptureClassTestCase(BaseScalyrLogCaptureTestCase):
+    """
+    Test class which verified BaseScalyrLogCaptureTestCase works correctly.
+    """
+    def test_assertLogFileContainsRegex(self):
+        # Log files should have been created in setUp()
+        self.assertTrue(os.path.isfile(self.agent_log_path))
+        self.assertTrue(os.path.isfile(self.agent_debug_log_path))
+
+        # Write some data to the file and verify assertLogFileContainsRegex works correctly
+        with open(self.agent_log_path, 'w') as fp:
+            fp.write(r'line 1\line 2\line 3')
```

Comment:
Did you mean to put newline characters in here so that they are actually separate lines?

Also, any particular reason you are using raw strings?  With the 2->3 migration, we are trying to be careful with our string usage.

Reply:
Yeah, that's a mistake on my part - there is no need to use a raw string here and there should indeed be a new line characters.

---

On a somewhat related note, I implemented ``assertLogFileContainsRegex`` so it mimics implementation from an existing test so it's line oriented.

To support also checks against the whole file content I also added ``assertLogFileContainsRegex``.

It's possible to also do line checks with a second more "generic" method so if people think it's too complicated or confusing to have two methods, we can just have one which performs checks against the content from the whole file.